paddle/framework/net_design.md

```diff
@@ -0,0 +1,72 @@
+# Network Design
+
+`Network` is the container and controller of a set of operators in a network, users can use `Network.AddOp` to add operators into a network, 
+and use `Network.Run()` to run all the operators in the network.
+
+The `Network` will
+
+- manage all the operators contained in the network.
+- not own any `Variable`.
+
+# API
+
+To make the `Network` extendibe, a base class is defined like this
+
+```c++
+// The minimum a network should be implemented.
+class NetworkBase {
+public:
+  BaseNetwork(const NetDef &def, Scope *scope);
```

Comment:
https://github.com/PaddlePaddle/Paddle/pull/2547#discussion_r123279210
If Network won`t hold scope, why we feed a scope to constructor?